extensions/attention-stream/src/youtube/utils/extractRecommendations.ts

```diff
@@ -0,0 +1,90 @@
+import * as traverse from "./traverseFishing";
+
+const extractRecommendations = ({ body }) => {
+  const renderers = traverse.fishForAll(
+    ["compactVideoRenderer", "videoRenderer" /*, "reelItemRenderer"*/], // leave out Reels/Shorts for now
+    body
+  );
+  if (renderers.length > 0) {
+    const detailsList = renderers
+      .filter(hasVideoId)
+      .map(sanitizeRecommendation);
+    const isInitialSet = !traverse.fishFor("continuationItems", body);
+    return {
+      detailsList,
+      isInitialSet,
+      firstTwentyVideoIds: isInitialSet
+        ? detailsList.map((rec) => rec.videoId).slice(0, 20)
+        : undefined,
+    };
+  }
+  return null;
+};
+
+export default extractRecommendations;
+
+// Some compactVideoRenderers have no top-level videoId
+// these are not algorithmic recommendations and can be ignored
+const hasVideoId = (rec) => rec.videoId;
+
+const sanitizeRecommendation = (rec) => {
+  // Remove fields that are purely for UI
+  traverse.fishForAndDeleteAll(["menu", "thumbnailOverlays"], rec);
+
+  // Extract useful details
+  // via nested destructuring
+  // (requires lots of default assignments)
+  const {
+    videoId,
+    descriptionSnippet: {
+      runs: [{ text: descriptionSnippet = undefined }] = [{}],
```

Comment:
Why undefined? This type is hard to comprehend since descriptionSnippet has an array which also has descriptionSnippet!